versions/3.1.0.md

```diff
@@ -2294,46 +2299,50 @@ The following properties are taken directly from the JSON Schema definition and
-- additionalProperties - Value can be boolean or object. Inline or referenced schema MUST be of a [Schema Object](#schemaObject) and not a standard JSON Schema. Consistent with JSON Schema, `additionalProperties` defaults to `true`.
+- propertyNames
+- contains
+- const
+- examples
+- if / then / else
+- dependentRequired / dependentSchemas
+- deprecated
+- additionalItems
+- additionalProperties
+- unevaluatedItems
+- unevaluatedProperties
+- maxContains
+- minContains
+- readOnly
+- writeOnly
+
+The following properties are taken from the JSON Schema definition but their definitions were adjusted to the OpenAPI Specification.
+
+- exclusiveMinimum - Value can be a numeric value just like JSON Schema, or for backwards compatibility with older drafts it can accept a boolean to act as a modifier to the `minimum` keyword. The numeric value usage is recommended.
+- exclusiveMaximum - Value can be a numeric value just like JSON Schema, or for backwards compatibility with older drafts it can accept a boolean to act as a modifier to the `minimum` keyword. The numeric value usage is recommended.
 - description - [CommonMark syntax](http://spec.commonmark.org/) MAY be used for rich text representation.
 - format - See [Data Type Formats](#dataTypeFormat) for further details. While relying on JSON Schema's defined formats, the OAS offers a few additional predefined formats.
-- default - The default value represents what would be assumed by the consumer of the input as the value of the schema if one is not provided. Unlike JSON Schema, the value MUST conform to the defined type for the Schema Object defined at the same level. For example, if `type` is `string`, then `default` can be `"foo"` but cannot be `1`.
+- default - The value MUST conform to the defined type for the Schema Object defined at the same level. For example, if `type` is `string`, then `default` can be `"foo"` but cannot be `1`. This differs from JSON Schema which only RECOMMENDS the default conforms to the schema.
 
 Alternatively, any time a Schema Object can be used, a [Reference Object](#referenceObject) can be used in its place. This allows referencing definitions instead of defining them inline.
 
-Additional properties defined by the JSON Schema specification that are not mentioned here are strictly unsupported.
-
-Other than the JSON Schema subset fields, the following fields MAY be used for further schema documentation:
+In addition to the JSON Schema fields, the following OpenAPI vocabulary fields MAY be used for further schema documentation:
 
 ##### Fixed Fields
 Field Name | Type | Description
 ---|:---:|---
-<a name="schemaNullable"></a>nullable | `boolean` | Allows sending a `null` value for the defined schema. Default value is `false`.
+<a name="schemaNullable"></a>nullable | `boolean` | Allows sending a `null` value for the defined schema. Default value is `false`. Deprecated. Move towards using `"type": "null"` & type arrays , e.g: `"type": ["string", "null"]`.
```

Comment:
Surely `"type": "null"` is not a useful construct.  Wouldn't we just recommend type arrays? This also means we should provide an annotation so that code generators would know which type to default for languages that don't support union types.

Reply:
This is talking about how “null” is added to the list of available types, and adding type arrays. If anyone would ever use type: null or not is fairy irrelevant and restricting it would probably be arbitrary.

Reply:
> Surely "type": "null" is not a useful construct. Wouldn't we just recommend type arrays?

Agree here, IMHO it makes sense to recommend usage of type arrays, even if a given definition starts off with only a single type in the array.

> This also means we should provide an annotation so that code generators would know which type

One option is to recommend leveraging the first type in the array in such cases - would that be too transparent? This is similar to what unofficially goes on in many tools around the `tags` array - the first tag is often used as the primary grouping mechanism, or the generated class name, etc.

Reply:
@darrelmiller @philsturgeon I think maybe re-wording this to emphasize `"null"` as a valid type that can be used in the array form to replace `nullable`, rather than writing out `"type": "null"`, would be less confusing.  (FWIW, I've seen `"type": "null"`used in some sort of weird compatibility mode where a field couldn't be entirely removed but no longer had any possible useful value- definitely a pathological case, but it can happen).

Reply:
Got it, great idea.

Reply:
@handrews @marbemac @darrelmiller I improved this confusing type: null working, please see if this is better or if I made things worse. I'm gonna click resolve so please comment on the new stuff.